lib/connection_controller.go

```diff
@@ -158,6 +166,21 @@ func (cc *ConnectionController) startNonValidatorConnector() {
 	}
 }
 
+func (cc *ConnectionController) startRemoteNodeCleanup() {
+	cc.startGroup.Done()
+
+	for {
+		select {
+		case <-cc.exitChan:
```

Comment:
Same comment in the other PRs. Does this work when multiple event loops are reading from the exitChan?